playbooks/roles/request-header-identity-provider/defaults/main.yml

```diff
@@ -0,0 +1,8 @@
+---
+
+# Default names for CR objects and user 
+default_ca_configmap_name: "ca-config-map"
+default_request_header_cr_name: "request-header-idp"
+default_request_header_username: "rh-user"
+default_request_header_display_name: "Default User"
+default_user_role: 'cluster-reader'
```

Comment:
Why is `default` in the variable names? This default file should have actual variable name with default values that are used when actual variable is undefined while running the role.

Reply:
As the default variables has lower precedence than the variables passed with `-e <file>`, hence not renamed these default variable names with the actual variables names. Instead added the fact variables at play level with the default values.